File: src/test/java/com/github/yuppieflu/stats/service/bucket/ManagedClock.java
package com.github.yuppieflu.stats.service.bucket;

import lombok.AllArgsConstructor;
import lombok.Setter;

import java.time.Clock;
import java.time.Instant;
import java.time.ZoneId;

@AllArgsConstructor
@Setter
class ManagedClock extends Clock {

    private Clock clock;

    @Override
    public ZoneId getZone() {
        return clock.getZone();
    }

    @Override
    public Clock withZone(ZoneId zone) {
        return clock.withZone(zone);
    }

    @Override
    public Instant instant() {
        return clock.instant();
    }
}

File: src/test/java/com/github/yuppieflu/stats/service/bucket/ManagedClock.kt
package com.github.yuppieflu.stats.service.bucket

import java.time.Clock
import java.time.ZoneId

class ManagedClock(var clock: Clock) : Clock() {
    override fun withZone(zone: ZoneId?) = clock.withZone(zone)

    override fun getZone() = clock.zone

    override fun instant() = clock.instant()
}
